cmake/boilerplate.cmake

```diff
@@ -28,3 +28,31 @@ if(DEFINED NRF_SUPPORTED_BUILD_TYPES)
                 message(FATAL_ERROR "${CMAKE_BUILD_TYPE} variant is not supported")
         endif()
 endif()
+
+string(FIND "${BOARD}" "@" REVISION_SEPARATOR_INDEX)
+if(NOT (REVISION_SEPARATOR_INDEX EQUAL -1))
+  math(EXPR BOARD_REVISION_INDEX "${REVISION_SEPARATOR_INDEX} + 1")
+  string(SUBSTRING ${BOARD} ${BOARD_REVISION_INDEX} -1 BOARD_REVISION)
+  string(SUBSTRING ${BOARD} 0 ${REVISION_SEPARATOR_INDEX} BOARD)
+endif()
+
+if(EXISTS ${CMAKE_SOURCE_DIR}/configuration)
+
+  if((DEFINED CONF_FILE) AND ("${CONF_FILE}" MATCHES "^configuration/${BOARD}/"))
+    # We have a relative conf file pointing inside configuration/<BOARD> folder
+    # hence old custom scheme is in use, thus do nothing until all samples have
+    # been updated to new scheme.
+  else()
+    set(APPLICATION_SOURCE_DIR ${CMAKE_SOURCE_DIR}/configuration/${BOARD}
```

Comment:
Let us use `config` instead of `configuration`.

Reply:
Alternative:
```
app/boards/nrf52840dk_nrf52840/{prj_release.conf,prj_debug.conf}
```

Reply:
I am for "boards".
Earlier discussed it with @tejlmand. I think that it will fit nicely as all files in the directory are board specific application files.

Reply:
> I am for "boards".

Seems reasonable. Would be nice to get this upstream though.

Reply:
completely agree, so will do the change and if no side effects are noticed we can send an upstream patch.

Reply:
> completely agree, so will do the change and if no side effects are noticed we can send an upstream patch.

My feeling is we should try to upstream this before merging it downstream if possible, but I won't block over it.

Reply:
@mbolivar-nordic thanks. I would like to have it cleaned up for NCS1.7 but for that to happen we need to finish the whole rework before mid August. I think it's not the best idea to do this change in rush around the release.
If the change cannot enter upstream quickly (maybe it can, I don't know), I would recommend to have it in downstream asap. Once it's in we would need to switch machine learning and nrf_desktop. In all cases we need to update the documentation. All of this takes time.

Reply:
I now have done the changes here:
https://github.com/nrfconnect/sdk-zephyr/pull/580/commits/ce90123da29ae723fdc55063dbc416476e12ec6c

@carlescufi @mbolivar-nordic could one of you take a quick look and see if this is in line with your expectations.
If yes, I will make an upstream PR and change title to `[nrf fromlist]`

Reply:
> @carlescufi @mbolivar-nordic could one of you take a quick look

I only read the commit message, but it like a good enough proposal to open a PR upstream.

Reply:
> In a private conversation @tejlmand expressed that we should not use boards for this purpose.

Correct, cause I feel that this messes up the boards folder in the app, and the description originates from the days before we actually used the boards folder for dt overlays and Kconfig fragments.
Now we have Zephyr modules in place which support a module.yml and thus easy integration of custom board to the project with automatic broader availability then I think this doc should be updated.

But following the referred docs will not impact whatever choices we make in this PR.